app/src/main/java/com/example/dinhh/soundscape/device/AddSound.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.dinhh.soundscape.device
+
+import android.media.MediaPlayer
+import com.example.dinhh.soundscape.data.Model
+import com.example.dinhh.soundscape.data.entity.SoundscapeItem
+import io.reactivex.Completable
+import java.io.IOException
+
+interface AddSound {
+    fun addSound(selectedSound: String, title: String, length: Int, category: String, volume: Int): Completable
+}
+
+class AddSoundImpl: AddSound{
+
+    override fun addSound(selectedSound: String, title: String, length: Int, category: String, volume: Int): Completable {
+        return Completable.create {
+
+            try {
+
+                val thisMp = MediaPlayer()
+                thisMp!!.reset()
+                thisMp.setDataSource(selectedSound)
+                thisMp.prepare()
+
+                val soundScapeItem = SoundscapeItem(title, length, thisMp, category, volume)
+
+                Model.selectedSounds.add(soundScapeItem)
+                it.onComplete()
+            } catch (e: IOException) {
+                it.onError(e)
+            }
+        }
+    }
+
```

Comment:
Your idea is that you add the selected sounds to an array/list and play them (I call it soundScapeList). 

- Add sound to a list is something very quick (not like you add it to database or to the server, no request required), you can savely remove AddSelectedLoading and  AddSelectedSoundSuccess. It is 99% success if you just add item to a list

- SoundscapeItem is an entity in data layer. Data layer is inner layer and it has no idea about the android world (it should not know what is MediaPlayer)

You can have that list in UI level (you creat a soundScapeList in SoundFragment and pass that list to the mixer via MainActivity). Avoid using global singleton like Model.kt